### PR TITLE
Fix cardinality inference on operators

### PR DESCRIPTION
**Pull request by elprans**
The operator cardinality inference currently doesn't take into account
the optionality of the return type, which results in incorrect lower
cardinality bound for most operators.  Function inference handles this
correctly.

Fixes: #2001

### CODE REVIEW

**Review by 1st1 — APPROVED** [review]

File: tests/test_schema.py [hunk 1]
```diff
@@ -5561,7 +5561,7 @@ def test_schema_describe_builtins_01(self):
                 CREATE OPTIONAL MULTI LINK intersection_of
                     -> schema::ObjectType;
                 CREATE OPTIONAL MULTI LINK union_of -> schema::ObjectType;
-                CREATE OPTIONAL SINGLE PROPERTY is_compound_type := (
+                CREATE REQUIRED SINGLE PROPERTY is_compound_type := (
```
Comment (1st1) [inline tests/test_schema.py:5564]: Is this related to the change?

**Review by msullivan — APPROVED** [review]

**Review by tailhook — APPROVED** [review]
